### DISCUSSION TITLE
Affichage de l'avis d'enregistrement

### DISCUSSION THREAD

**Discussion by jodionrq** [MTESSDev/FRW#105] [Annonces]
Bonjour,
L'avis "afficherMessageIncitatif" apparait bien dans nos formulaires lorsqu'on les ouvrent dans le Bac-à-sable, mais ne s'affiche pas lorsque le formulaire est gradué dans l'environnement de démarrage d'essai. Il s'agit de l'avis suivant :
![image](https://github.com/MTESSDev/FRW/assets/124822406/1302a7aa-91c2-4415-b76d-56eb4d84cb87)

Le bouton d'enregistrement au bas des pages quant à lui s'affiche et fonctionne bien.
Est-ce normal?

**Comment by LarcM7** [comment]
Bonjour, c'est peut-être seulement dû à la façon de créer le formulaire.

- Si vous utilisez le champ **Données en entrée (pré-remplissage)**, le système considère que c'est un formulaire authentifié, et le comportement attendu est de ne pas afficher l'avis;
- Si vous cochez l'option **Accès anonyme**, le système considère que c'est un formulaire anonyme, alors l'avis "afficherMessageIncitatif" sera affiché.

Voulez-vous essayer ça et me revenir en me disant si cela répond à votre question? où si vous avez d'autres questions ou commentaires, n'hésitez pas.

Merci

**Reply by jodionrq** [reply]
Merci pour la réponse.
Présentement nos fichiers sont programmés de cette façon :
Dans la partie "config" :
````yaml
  enregistrement:
    actif: true
    afficherMessageIncitatif: true
````
Dans la partie où débute le formulaire :
````yaml
accesAnonyme: true
afficherMessageSauvegardeCourrielReprise: false
````
Nous utilisons l'option des données en pré-remplissage.
Si on coche "Accès anonyme" ça me dit que le formulaire n'est pas disponible.
Est-ce que ça veut dire que je devrais changer la ligne accesAnonyme pour "false" pour que le message s'affiche dans notre cas?

**Reply by LarcM7** [reply]
Bonjour,

À première vue je ne comprenais pas, car selon ces informations tout devrait fonctionner. Puis j'ai regardé les configs de vos formulaires, et j'ai vu deux problèmes avec celui-ci RDC-025-001-STA.v1.form.yml : 

- la clé accesAnonyme n'est pas bien indentée (ligne 86 (et aussi 87))
- la clé accesAnonyme n'est pas sous la clé "securite"

Vous trouverez un exemple ici : https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=personnalisationBase

Si ce n'est pas ce formulaire qui est en problème, je vais devoir poursuivre mes recherches.

Pouvez-vous vérifier et me revenir?

Merci

**Reply by anisite** [reply]
Bonsoir, 

Juste une mini précision pour votre config, cette partie: 
![image](https://github.com/MTESSDev/FRW/assets/3194288/6b9e4da5-8a06-4790-a457-bea95f3775ca)

C'est un ancien pattern qui fonction uniquement lorsqu'il n'y a pas de bloc config: dès qu'on ajoute ce bloc le traitement se comporte avec le nouveau pattern.

assurez-vous aussi d'éditer avec VS Code et d'avoir comme première ligne dans votre fichier de config de formulaire cette ligne :
````yaml
# yaml-language-server: $schema=https://formulaires.it.mtess.gouv.qc.ca/api/v1/SIS/ObtenirSchema/form
````
c'est elle qui est responsable de valider la structure de votre fichier, normalement elle affiche les erreurs dans celui-ci

**Reply by jodionrq** [reply]
Bonjour,
Merci pour la précision. En effet au début de l'utilisation de FRW nous n'avions pas de partie "config".
Donc, si nous voulons que ce formulaire passe par une authentification dans Mon dossier, j'imagine que l'accesAnonyme sera à "false" ?
Je vais faire un essai lors d'une prochaine graduation pour voir si le message incitatif de sauvegarde s'affiche bien.
Merci !

**Reply by anisite** [reply]
Bonsoir,

En fait comme @LarcM7 le précisait, l'incitation à la sauvegarde ne fonctionne qu'en monde anonyme puisque en mode authentifié la sauvegarde est automatique et qu'en tout temps l'utilisateur pourra reprendre son formulaire à l'état où il l'a laissé depuis son dossier en ligne. 

Le message est donc inutile puisqu'il a été prévu pour ceux qui risquent de perdre leur formulaire en n'ayant pas d'espace client pour le reprendre par la suite, la sauvegarde anonyme envoie un lien par courriel pour permettre le retour dans le formulaire incomplet.
